src/PerformanceSensitiveAnalyzers/CSharp/CodeFixes/AvoidAllocationWithArrayEmptyCodeFix.cs

```diff
@@ -0,0 +1,283 @@
+            }
+        }
+
+        private static bool IsMethodInvocationParameter(SyntaxNode node) => node is ArgumentSyntax;
+
+        private static bool IsReturnStatement(SyntaxNode node)
+        {
+            return node.Parent is ReturnStatementSyntax or YieldStatementSyntax or ArrowExpressionClauseSyntax;
+        }
+
+        private static bool IsInsideMemberReturningEnumerable(SyntaxNode node, SemanticModel semanticModel)
+        {
+            return IsInsideMethodReturningEnumerable(node, semanticModel) ||
+                   IsInsidePropertyDeclaration(node, semanticModel);
+
+        }
+
+        private static bool IsInsidePropertyDeclaration(SyntaxNode node, SemanticModel semanticModel)
+        {
+            var propertyDeclaration = node.FirstAncestorOrSelf<PropertyDeclarationSyntax>();
+            return propertyDeclaration != null &&
+                IsPropertyTypeReadonlySequence(semanticModel, propertyDeclaration) &&
+                (IsAutoPropertyWithGetter(node) || IsArrowExpression(node));
+        }
+
+        private static bool IsAutoPropertyWithGetter(SyntaxNode node)
+        {
+            var accessorDeclaration = node.FirstAncestorOrSelf<AccessorDeclarationSyntax>();
+            return accessorDeclaration != null && accessorDeclaration.Keyword.Text == "get";
+        }
+
+        private static bool IsArrowExpression(SyntaxNode node)
+        {
+            return node.FirstAncestorOrSelf<ArrowExpressionClauseSyntax>() != null;
+        }
+
+        private static bool CanBeReplaceWithEnumerableEmpty(ArrayCreationExpressionSyntax arrayCreation)
+        {
+            return IsInitializationBlockEmpty(arrayCreation.Initializer);
+        }
+
+        private static bool CanBeReplaceWithEnumerableEmpty(ObjectCreationExpressionSyntax objectCreation, SemanticModel semanticModel)
+        {
+            return IsCollectionType(semanticModel, objectCreation) &&
+                   IsInitializationBlockEmpty(objectCreation.Initializer) &&
+                   IsCopyConstructor(semanticModel, objectCreation) == false;
+        }
+
+        private static bool IsInsideMethodReturningEnumerable(SyntaxNode node, SemanticModel semanticModel)
+        {
+            var methodDeclaration = node.FirstAncestorOrSelf<MethodDeclarationSyntax>();
+            return methodDeclaration != null && IsReturnTypeReadonlySequence(semanticModel, methodDeclaration);
+        }
+
+        private static async Task<Document> Transform(Document contextDocument, SyntaxNode node, TypeSyntax typeArgument, CancellationToken cancellationToken)
+        {
+            var noAllocation = SyntaxFactory.ParseExpression($"Array.Empty<{typeArgument}>()");
+            var newNode = ReplaceExpression(node, noAllocation);
+            if (newNode == null)
+            {
+                return contextDocument;
+            }
+            var syntaxRootAsync = await contextDocument.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
[minor] No code style is applied on the repo so feel free to ignore this but I would recommend a blank line after the if.

```suggestion

            var syntaxRootAsync = await contextDocument.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
```